spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -16,114 +16,119 @@ This standard document specifies packet data structure, state machine handling l
 
-`IBCAccountPacketData` specifies the type of packet. The `IBCAccountPacketData.data` is arbitrary and can be used for different purposes depending on the value of `IBCAccountPacketData.type`.
+`runTx` executes a transaction based on the IBC packet recieved from the controller chain.
 
-When the value of `type` is `REGISTER`, `IBCAccountPacketData.data` can be used as information by which the account is created on the host chain. When the value of `type` is `RUNTX`, `IBCAccountPacketData.data` contains the tx bytes.
+### Packet Data
+`InterchainAccountPacketData` contains an array of messages that an interchain account can execute and a memo string that is sent to the host chain.  
 
 ```typescript
-// `Type` enumerator defines the packet type.
-enum Type {
-  REGISTER,
-  RUNTX
-}
-
-interface IBCAccountPacketData {
-  type: Type
-  data: Uint8Array
+message InterchainAccountPacketData  {
+    repeated google.protobuf.Any messages = 1;
+    string memo = 2;
 }
 ```
 
-The acknowledgement data type describes the type of packet data and chain id and whether the result code of processing and the data of result if it is needed, and the reason for failure (if any).
+The acknowledgement packet structure is defined as in [ics4](https://github.com/cosmos/cosmos-sdk/blob/v0.42.4/proto/ibc/core/channel/v1/channel.proto#L134-L147). If an error occurs on the host chain the acknowledgement should contain the error message.
 
 ```typescript
-interface IBCAccountPacketAcknowledgement {
-    type: Type
-    code: uint32
-    data: Uint8Array
-    error: string
+message Acknowledgement {
+  // response contains either a result or an error and must be non-empty
+  oneof response {
+    bytes  result = 21;
+    string error  = 22;
+  }
 }
 ```
 
-The ```IBCAccountHook``` interface allows the source chain to receive results of executing transactions on an interchain account.
-
+The ```InterchainAccountHook``` interface allows the controller chain to receive results of executed transactions on the host chain.
 ```typescript
-interface IBCAccountHook {
-  onAccountCreated(sourcePort:string, sourceChannel:string, address: Address)
+interface InterchainAccountHook {
   onTxSucceeded(sourcePort:string, sourceChannel:string, txBytes: Uint8Array)
   onTxFailed(sourcePort:string, sourceChannel:string, txBytes: Uint8Array)
 }
 ```
 
-### Subprotocols
-
-The subprotocols described herein should be implemented in an "interchain-account-bridge" module with access to a router and codec (decoder or unmarshaller) for the application and access to the IBC relayer module.
-
 ### Port & channel setup
+The interchain account module on a host chain must always bind to a port with the id `interchain_account`. Controller chains will bind to ports dynamically, with each port id set as `ics27-1-{owner-address}`.
```

Comment:
Could we make the port naming a bit more obvious? `interchain_account_host`, `interchain_account_control_{owner-address}`

Or replace `interchain_account` with `ics27-1` in both cases